spec/system/home_page_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.feature "Home Page" do
+  scenario "User visits the claims homepage" do
+    given_i_am_on_the_claims_site
+    and_i_am_on_the_start_page
```

Comment:
Can you remove these type of private methods. I guess there is a config aspect to set the host, maybe use something like
```
RSpec.describe "Widget management", type: :request do
  before { host! "api.example.com" }

  it "creates a Widget" do
    headers = { "ACCEPT" => "application/json" }
    post "/widgets", :params => { :widget => { :name => "My Widget" } }, :headers => headers

    expect(response.content_type).to start_with("application/json")
    expect(response).to have_http_status(:created)
  end
end
```
, or a global config

If you want your test to be more expressive you can make a use of `describe|context|scenario|it`.

Good work

Reply:
I've referenced the existing BAT apps for the system specs.

[Apply example](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/spec/system/candidate_interface/candidate_content_spec.rb)
[Register example](https://github.com/DFE-Digital/register-trainee-teachers/blob/main/spec/features/landing_page_spec.rb)

That being said, I'm not against the format you're proposing. What do you guys think? @DFE-Digital/track-pay @DFE-Digital/school-placements

Reply:
Weighing in --- I actually prefer the way this is currently written. When we have good ACs on tickets, this pattern maps straight to the ACs, but without the nightmare of using something like cucumber. And if it's the pattern used more widely across the DfE, it makes sense to continue it here. 

Happy to go with the majority if other don't agree, of course.

Reply:
I didn't realise other repo where doing this. Looks a bit crazy but my main concern is this:

- Usually you want very low amount coding in your test and accept some level of duplication. If you code in your test you now need to test this code... So it's important(to me at least) to be able to spot quickly where is the code(aka magic) which sometimes is ok. This will create a situation where a LOT of functions may or may not do very similar things. 

But if the team feel like it's ok to go with a lot of small functions I will follow.

Reply:
I think we'd be best to stick with the convention that's already established here in DfE.

For anyone who hasn't seen this style of feature spec before, it's worth pointing out that it does have a small but established following. It's not invented here at DfE.

Here's a blog post describing the style and the reasoning behind it:
_(via the Wayback Machine because the live URL seems to be broken)_
https://web.archive.org/web/20181125211832/https://about.futurelearn.com/blog/how-we-write-readable-feature-tests-with-rspec/

This seems like a nice way to ensure we create readable feature specs.